tests/009-close.cpp

```diff
@@ -0,0 +1,52 @@
+#include <catch2/catch_test_macros.hpp>
+#include <quic.hpp>
+#include <quic/gnutls_crypto.hpp>
+#include <thread>
+
+#include "utils.hpp"
+
+namespace oxen::quic::test
+{
+    using namespace std::literals;
+
+    TEST_CASE("009 - Close", "[009][close][callbacks]")
+    {
+        uint64_t client_error{0};
+        uint64_t server_error{0};
+        auto client_established = bool_waiter{[](connection_interface&) {}};
+        auto server_established = bool_waiter{[](connection_interface&) {}};
+        // this needs to be destroyed *after* Network, as it may be called during ~Network
+        auto client_closed = bool_waiter{[&client_error](connection_interface&, uint64_t ec) { client_error = ec; }};
+        auto server_closed = bool_waiter{[&server_error](connection_interface&, uint64_t ec) { server_error = ec; }};
+
+        Network test_net{};
+
+        auto server_tls = GNUTLSCreds::make("./serverkey.pem"s, "./servercert.pem"s, "./clientcert.pem"s);
+        auto client_tls = GNUTLSCreds::make("./clientkey.pem"s, "./clientcert.pem"s, "./servercert.pem"s);
+
+        opt::local_addr server_local{};
+        opt::local_addr client_local{};
+
+        SECTION("Connection Closes with correct error code")
```

Comment:
looks good to me! one question: are you planning on adding a second section to this te3st case?

Reply:
nah, just couldn't remember if it had to have a section or not and figured it wouldn't hurt to put it regardless

Reply:
agreed